src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -1100,37 +1150,198 @@ private void PopulateParamsFromRestRequest(Dictionary<string, object?> parameter
+        ///                         title: "book #2",
+        ///                         reviews: [{ content: "Awesome book." }, { content: "Average book." }],
+        ///                         publisher: { name: "Pearson Education" },
+        ///                         authors: [{ birthdate: "1990-11-04", name: "Penguin Random House", author_name: "William Shakespeare" }]
+        ///                     }])
+        ///                 {
+        ///                     items{
+        ///                         id
+        ///                         title
+        ///                     }
+        ///                 }</example>
+        private void PopulateMutationEntityAndFieldsToAuthorize(
+            Dictionary<string, HashSet<string>> entityToFields,
+            IInputField schema,
+            string entityName,
+            IMiddlewareContext context,
+            object? parameters)
         {
-            string role = string.Empty;
-            if (context.ContextData.TryGetValue(key: AuthorizationResolver.CLIENT_ROLE_HEADER, out object? value) && value is StringValues stringVals)
+            if (parameters is List<ObjectFieldNode> listOfObjectFieldNode)
             {
-                role = stringVals.ToString();
+                // For the example createbook mutation written above, the object value for `item` is interpreted as a List<ObjectFieldNode> i.e.
+                // all the fields present for item namely- title, reviews, publisher, authors are interpreted as ObjectFieldNode.
+                ProcessObjectFieldNodesForAuthZ(
+                    context: context,
+                    entityName: entityName,
+                    schemaObject: ResolverMiddleware.InputObjectTypeFromIInputField(schema),
+                    fieldNodes: listOfObjectFieldNode,
+                    entityTofields: entityToFields);
             }
-
-            if (string.IsNullOrEmpty(role))
+            else if (parameters is List<IValueNode> listOfIValueNode)
+            {
+                // For the example createbooks mutation written above, the list value for `items` is interpreted as a List<IValueNode>
+                // i.e. items is a list of ObjectValueNode(s).
+                listOfIValueNode.ForEach(iValueNode => PopulateMutationEntityAndFieldsToAuthorize(
+                    entityToFields: entityToFields,
+                    schema: schema,
+                    entityName: entityName,
+                    context: context,
+                    parameters: iValueNode));
+            }
+            else if (parameters is ObjectValueNode objectValueNode)
+            {
+                // For the example createbook mutation written above, the node for publisher field is interpreted as an ObjectValueNode.
+                // Similarly the individual node (elements in the list) for the reviews, authors ListValueNode(s) are also interpreted as ObjectValueNode(s).
+                ProcessObjectFieldNodesForAuthZ(
```

Comment:
super nit: any way that this method `ProcessObjectFieldNodesForAuthZ(...)` can have the same param order as `PopulateMutationEntityAndFieldsToAuthorize(...)`

```
context: 
entityName:
schemaObject:
fieldNodes:
entityTofields: 
```

vs 

```
entityToFields: ,
schema: ,
entityName: ,
context: 
parameters: )
```